Week-1/assignment1/readme.md

```diff
@@ -0,0 +1,60 @@
+
+![rendertree](images/rendertree.png)
+
+### G – Order of script processing
+
+Explanation – JavaScript on web page executes based on certain factors and orders, let's consider where and how to add JavaScript to a web page There are basically three locations into which we can attach JavaScript: • Directly into the head of the page • Directly into the body of the page • From an event handler/listener Directly into the head of the page and body of the page: What does it mean to say that JavaScript is directly in the head or body of the page? If the code is not enclosed in a function or object, it is directly in the page. In this case, the code runs sequentially as soon as the file containing the code has loaded sufficiently for that code to be accessed. Code that is within a function or object is run only when that function or object is called. Basically, this means that any code inside the head and body of your page that is not inside a function or object will run as the page is loading — as soon as the page has loaded sufficiently to access that code. That last bit is important and impacts the order in which you place your code on the page: any code placed directly in the page that needs to interact with elements within the page must appear after the elements in the page on which it is dependent. In general, this means that if you use direct code to interact with your page content, such code should be placed at the bottom of the body. Code Assigned to Event Handlers and Listeners Assigning a function to an event handler or listener does not result in the function being run at the point at which it is assigned — provided that you are actually assigning the function itself and not running the function and assigning the value returned. (This is why you generally do not see the () on the end of the function name when it is being assigned to an event since the addition of the parentheses runs the function and assigns the value returned rather than assigning the function itself.) Functions that are attached to event handlers and listeners run when the event that they are attached to is triggered. Most events are triggered by visitors interacting with your page. Some exceptions exist, however, such as the load event on the window itself, which is triggered when the page finishes loading. Functions Attached to Events on Page Elements Any functions attached to events on elements within the page itself will run according to the actions of each individual visitor — this code runs only when a particular event occurs to trigger it. For this reason, it doesn't matter if the code never runs for a given visitor, since that visitor has obviously not performed the interaction that requires it. All of this, of course, assumes that your visitor has accessed your page with a browser that has JavaScript enabled.
+
```

Comment:
Like mentioned earlier...

Please keep it short and sweet.

Break it down in to paragraphs, bulleted lists etc... That will make it readable.

Reply:
Please use more markdown features to make it easier for readers to understand your explanation...

Otherwise it is difficult.

Reply:
Vinit, I feel that you've went into unnecessary details in this part. This is good... but like I said... please use shorter paragraphs, bulleted lists... 

My understanding from the assignment is....
We are supposed to write a high level overview of how a browser works... That's all...
Please summarize this in 5-8 sentences. That should be more than enough for us.